plugin/kubernetes/handler_test.go

```diff
@@ -408,6 +408,14 @@ var dnsTestCases = []kubeTestCase{
 			test.SOA("cluster.local.	5	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1499347823 7200 1800 86400 5"),
 		},
 	}},
+	// A query for a double subdomain of an external service should not resolve to the external service
+	{Case: test.Case{
+		Qname: "double.endpoint.external.testns.svc.cluster.local.", Qtype: dns.TypeCNAME,
```

Comment:
I started with only setting the port/protocol pseudo segments if they started with an underscore but then went with this other approach because it could also effect headless service lookups. I can revisit this approach if it's preferred. hopefully some others can chime in on the SRV lookup questions.